ext/http/01_http.js

```diff
@@ -74,104 +75,82 @@ class HttpConn {
   #closed = false;
   #remoteAddr;
   #localAddr;
-
-  // This set holds resource ids of resources
-  // that were created during lifecycle of this request.
-  // When the connection is closed these resources should be closed
-  // as well.
-  managedResources = new Set();
+  #connections;
+  #waiters;
+  #httpRid;
+  #requests;
+  #error;
 
   constructor(rid, remoteAddr, localAddr) {
-    this.#rid = rid;
     this.#remoteAddr = remoteAddr;
     this.#localAddr = localAddr;
+    this.#connections = [];
+    this.#waiters = [];
+    this.#requests = new Set();
+
+    const connections = this.#connections;
+    const waiters = this.#waiters;
+    const context = new CallbackContext();
+    const onError = this.#onError.bind(this);
+
+    this.#httpRid = core.ops.op_serve_http_on(rid, (fallback_base_url) => context.initialize(fallback_base_url), map_to_userland(this.#requests, context, (req) => {
+      const promise = new Promise(r => {
+        connections.push([r, req]);
+        waiters.pop()?.(true);
+      });
+      return promise;
+    }, onError));
+    // Start the async IIFE to avoid looking like the event loop stalled
+    const httpRid = this.#httpRid;
+    (async () => {
+      try {
+        await core.opAsync("op_http_wait", httpRid);
+      } catch (e) {
+        onError(e);
+      }
+    })()
```

Comment:
This async IIFE can (and I think it will) cause resource and ops sanitizer to complain - I suggest to store it somewhere on the struct and await it when closing the server.